workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/DashboardPage/DashboardPage.tsx

```diff
@@ -71,6 +71,18 @@ export function DashboardPage() {
 		return productType;
 	}
 
+	function getProductVersionFromSpecifications(specifications: any) {
+		var productVersion = '0';
+
+		specifications.items.forEach((specification: any) => {
+			if (specification.specificationKey === "version") {
+				productVersion = specification.value.en_US;
+			}
+		})
+
+		return productVersion;
+	}
+
```

Comment:
Why not call for specifications of each product when looping through the appList?  Seems like we are looping through the list two more times than necessary and storing productIds when we don't have to.  What do you think?

Reply:
For the specifications, I've tried to get the specifications while looping through the app list before but since getting the specifications is another API call, it was pretty complicated to get it to work. Allen suggested that I try it the current way instead (even though it looks messy) by looping through the array of product IDs.